### PR TITLE
Site Transfers: Prototype UX to initiate a site transfer

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/Automattic/dotcom-forge/issues/2496

## Proposed Changes

Prototypes a workflow for a site owner to transfer their site to another WordPress.com user.

However, this will change substantially in a follow-up issue https://github.com/Automattic/dotcom-forge/issues/2496#issuecomment-1558897012

### New 'Transfer your site' option in general settings

image

### Initial screen

image

### Input appears once you've toggled the boxes

image

## Testing Instructions

1. Apply D111563-code to your sandbox
2. Create a new Simple site with a Personal plan using a test user.
3. Navigate to General Settings and click 'Transfer your site'.
4. Click on the toggles and verify the input appears.
5. Add an invalid username, hit submit, and verify an error message appears.
6. Add a username for a second test user, hit submit, and verify the success message appears.
7. Verify you've received a confirmation email in your inbox.

### CODE REVIEW

**Comment by katinthehatsite** [comment]
I know this is not ready for review and probably should be opened as different issues but wanted to briefly document it before I forget. The toggles are a bit cut off on the left side:

Screenshot 2023-05-23 at 11 16 08 AM

Once you toggle things on, there is a light grey line that somewhat overlaps the button:

Screenshot 2023-05-23 at 11 17 11 AM

Otherwise 😻

**Review by katinthehatsite — APPROVED** [review]
I tested the steps as outlined in the PR and was able to complete the transfer as described. The only comments were the small UI improvements from above. Otherwise, works as expected.

File: client/my-sites/site-settings/site-tools/index.jsx [hunk 1]
```diff
@@ -190,6 +205,9 @@ export default compose( [
 				sourceSlug: siteSlug,
 			} );
 
+			const showStartSiteTransfer =
+				isEnabled( 'yolo/site-transfers-i1' ) && ! isJetpack && ! isVip && ! isP2Hub && ! isAtomic;
```
Comment (sejas) [inline client/my-sites/site-settings/site-tools/index.jsx]: Should we check if the user is a site owner? Ideally, we could reuse a function like `canStartSiteOwnerTransfer` for this condition.

Now, non-owner admins that are not site owners can see the "CTA" but when the user clicks on it, is redirected to the same page - General Settings.

https://github.com/Automattic/wp-calypso/assets/779993/4ca9992c-c68a-44ac-b719-af5a1d39beff

## steps to reproduce
- Add a user as admin
- Accept the invitation
- With that non-owner admin go to Settings > General
- Click on `Transfer your site`

**Review by sejas — COMMENTED** [review]
Tested along with the diff and it works great.
There is a corner case where a non-owner tries to transfer the site.

**Comment by sejas** [comment]
Should we replace the word `blog` to `site`?

I can see inconsistencies with the title and the CTA `Start site transfer` and the input label `USERNAME OR EMAIL ADDRESS OF USER TO RECEIVE THE BLOG`. I would prefer to use the word `site`: _USERNAME OR EMAIL ADDRESS OF USER TO RECEIVE THE ~BLOG~ SITE_

Screenshot 2023-05-24 at 11 56 39

**Comment by danielbachhuber** [comment]
> Should we replace the word `blog` to `site`?

@sejas We'll adjust all of the copy in a follow-up issue: https://github.com/Automattic/dotcom-forge/issues/2496#issuecomment-1559430610

**Comment by sejas** [comment]
> I know this is not ready for review and probably should be opened as different issues but wanted to briefly document it before I forget. The toggles are a bit cut off on the left side:

@danielbachhuber , @katinthehatsite , I've created this PR to fix these two issues:
- Toggle overflow. Here I'm changing the style for all the `.action-panel__body`.
- Button overlapping with the line

- https://github.com/Automattic/wp-calypso/pull/77322

File: client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+			},
+		}
+	);
+
+	const handleFormSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+		startSiteOwnerTransfer( { newSiteOwner: newOwnerUsername } );
+	};
+
+	const startSiteTransferForm = (
```
Comment (wojtekn) [inline client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx:75]: What do you think about moving the form to the separate child component? Or if we are not going to reuse it anywhere, or use in storybook, maybe we can keep the form inline with the rest of component code?

Comment (danielbachhuber) [reply client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx:75]: @wojtekn The form is changing in a follow-up issue https://github.com/Automattic/dotcom-forge/issues/2496#issuecomment-1558897012

File: client/my-sites/site-settings/site-tools/index.jsx (hunk 1 above)
Comment (danielbachhuber) [reply client/my-sites/site-settings/site-tools/index.jsx]: Good catch, fixed with 96ee79f969

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8008449

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8008449).

**Pull request merged**